### PR TITLE
Fix Build

### PR DESCRIPTION
**Pull request by amitksingh1490**

### CODE REVIEW

File: example/src/main/scala/HelloWorld.scala [hunk 1]
```diff
@@ -6,7 +6,7 @@ import zio.stream._
 
 object HelloWorld extends App {
 
-  def h1 = HttpApp.fromHttp {
+  def h1 = HttpApp {
```
Comment (tusharmath) [inline example/src/main/scala/HelloWorld.scala]: unnecessary change

File: zio-http/src/main/scala/zhttp/experiment/Router.scala [hunk 2]
```diff
@@ -34,7 +36,7 @@ object Router {
     def /[B, C](other: Router[B])(implicit ev: Combine.Aux[Unit, B, C]): Router[C] = ???
   }
 
-  val route = Method.GET / "a" / "b" / ![Int] / "c" / ![String] / ![Int]
+  // val route = Method.GET / "a" / "b" / ![Int] / "c" / ![String] / ![Int]
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/experiment/Router.scala]:
```suggestion
  val route = Method.GET / "a" / "b" / ![Int] / "c" / ![String] / ![Int]
```

File: zio-http/src/test/scala/zhttp/experiment/internal/HttpGen.scala [hunk 3]
```diff
@@ -100,12 +100,13 @@ object HttpGen {
   def header = for {
     key   <- Gen.alphaNumericStringBounded(1, 4)
     value <- Gen.alphaNumericStringBounded(1, 4)
-  } yield Header(key, value)
+  } yield Header.custom(key, value)
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/experiment/internal/HttpGen.scala]: Why do we need Header constructor to be private?

File: zio-http/src/test/scala/zhttp/service/CORSSpec.scala [hunk 4]
```diff
@@ -43,11 +43,6 @@ object CORSSpec extends HttpRunnableSpec(8089) {
                 ),
               ),
             )
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/service/CORSSpec.scala:45]:
```suggestion
            ) &&
```

File: zio-http/src/test/scala/zhttp/service/ClientContentLengthSpec.scala [hunk 5]
```diff
@@ -13,7 +12,7 @@ object ClientContentLengthSpec extends HttpRunnableSpec(8083) {
 
   type ServerState = Map[String, Int]
 
-  val env = EventLoopGroup.auto() ++ ChannelFactory.auto ++ ServerChannelFactory.auto
+  val env = EventLoopGroup.auto(1) ++ ChannelFactory.auto ++ ServerChannelFactory.auto
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/service/ClientContentLengthSpec.scala]:
```suggestion
  val env = EventLoopGroup.auto() ++ ChannelFactory.auto ++ ServerChannelFactory.auto
```

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala [hunk 6]
```diff
@@ -7,7 +7,7 @@ import zio.test.Assertion.equalTo
 import zio.test.assertM
 
 object ServerSpec extends HttpRunnableSpec(8087) {
-  val env = EventLoopGroup.auto() ++ ChannelFactory.auto ++ ServerChannelFactory.auto
+  val env = EventLoopGroup.auto(1) ++ ChannelFactory.auto ++ ServerChannelFactory.auto
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/service/ServerSpec.scala]:
```suggestion
  val env = EventLoopGroup.auto(0) ++ ChannelFactory.auto ++ ServerChannelFactory.auto
```

File: zio-http/src/main/scala/zhttp/http/HttpApp.scala [hunk 7]
```diff
@@ -117,7 +114,7 @@ case class HttpApp[-R, +E](asHttp: Http[R, E, Request, Response[R, E]]) { self =
               unsafeWriteAnyResponse(a)
               a.data match {
                 case HttpData.Empty =>
-                  unsafeWriteAndFlushNotFoundResponse()
+                  unsafeWriteLastContent(Unpooled.EMPTY_BUFFER)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/HttpApp.scala]: Need to add tests in ResponseSpec

File: zio-http/src/main/scala/zhttp/experiment/Router.scala (hunk 2 above)
Comment (amitksingh1490) [reply zio-http/src/main/scala/zhttp/experiment/Router.scala:39]: In Scala 3 ! operator is not working renamed it to Arg, Also Combine is not working for 3 variables

File: zio-http/src/main/scala/zhttp/http/HttpApp.scala (hunk 7 above)
Comment (amitksingh1490) [reply zio-http/src/main/scala/zhttp/http/HttpApp.scala]: Done

File: zio-http/src/main/scala/zhttp/experiment/Router.scala [hunk 8]
```diff
@@ -16,16 +18,16 @@ object Router {
...
     implicit object StringExtract extends RouteParam[String] {
       override def extract(data: String): Option[String] = Option(data)
     }
   }
 
-  sealed trait ![A] extends Router[A]
-  object ! {
+  sealed trait Arg[A] extends Router[A]
+  object Arg {
     def apply[A](implicit ev: RouteParam[A]): Router[A] = ???
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/experiment/Router.scala]: move apply out of arg, inside `Router`

File: zio-http/src/main/scala/zhttp/http/HttpApp.scala [hunk 9]
```diff
@@ -165,15 +162,19 @@ case class HttpApp[-R, +E](asHttp: Http[R, E, Request, Response[R, E]]) { self =
                 ctx.read(): Unit
               }
 
-            case ContentDecoder.Custom(state, run) =>
+            case ContentDecoder.Custom(
+                  state: Any,
+                  runx,
+                ) =>
+              val run    = runx.asInstanceOf[((Chunk[Byte], Any, Boolean) => ZIO[R, Throwable, (Option[Any], Any)])]
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/http/HttpApp.scala]: rename

File: zio-http/src/main/scala/zhttp/http/HttpApp.scala [hunk 10]
```diff
@@ -165,15 +162,19 @@ case class HttpApp[-R, +E](asHttp: Http[R, E, Request, Response[R, E]]) { self =
...
               if (ad.isFirst) {
                 ad.decoderState = state
                 ad.isFirst = false
               }
               val nState = ad.decoderState
 
               unsafeRunZIO(for {
-                (publish, state) <- run(Chunk.fromArray(content.array()), nState, isLast)
+                (publish, state) <- run(Chunk.fromArray(ByteBufUtil.getBytes(content)), nState, isLast)
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/http/HttpApp.scala]: add test in a new PR

File: zio-http/src/test/scala/zhttp/service/CORSSpec.scala [hunk 11]
```diff
@@ -8,7 +8,7 @@ import zio.test.Assertion._
 import zio.test.assertM
 
 object CORSSpec extends HttpRunnableSpec(8089) {
-  val env = EventLoopGroup.auto() ++ ChannelFactory.auto ++ ServerChannelFactory.auto
+  val env = EventLoopGroup.auto(1) ++ ChannelFactory.auto ++ ServerChannelFactory.auto
```
Comment (amitksingh1490) [inline zio-http/src/test/scala/zhttp/service/CORSSpec.scala]: remove 1

File: zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala [hunk 12]
```diff
@@ -40,12 +40,17 @@ case class HttpAppClient(outbound: MessageQueue[HttpObject], channel: ProxyChann
...
 
   def end(iter: Iterable[String]): Task[Unit] = {
     if (iter.isEmpty) end
-    else ZIO.foreach(iter.zipWithIndex) { case (c, i) => writeText(c, isLast = i == iter.size - 1) }.unit
+    else
+      ZIO
+        .foreach(iter.zipWithIndex) { case (c, i) =>
+          writeText(c, isLast = i == iter.size - 1)
+        }
+        .as(())
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala:53]: Can u verify if this is required?

File: zio-http/src/test/scala/zhttp/service/ClientContentLengthSpec.scala [hunk 13]
```diff
@@ -54,20 +53,20 @@ object ClientContentLengthSpec extends HttpRunnableSpec(8083) {
...
               val actual  = request(!! / path, Method.POST, content) *> getLengthForPath(state, path)
               assertM(actual)(isSome(isPositive[Int]))
-            } +
-            testM("post request with nonempty content and set content-length") {
-              val path    = "postWithNonemptyContentAndSetContentLength"
-              val content = "content"
-              val headers = List(Header.custom(contentLengthName, "dummy"))
-              val actual  = request(!! / path, Method.POST, content, headers) *> getLengthForPath(state, path)
-              assertM(actual)(isSome(isPositive[Int]))
             },
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/service/ClientContentLengthSpec.scala]:
```suggestion
            } +
```

File: zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala (hunk 12 above)
Comment (amitksingh1490) [reply zio-http/src/test/scala/zhttp/experiment/internal/HttpAppClient.scala]: Not needed

**Review by tusharmath — APPROVED** [review]
